skopt/optimizer/optimizer.py

```diff
@@ -520,6 +520,31 @@ def tell(self, x, y, fit=True):
         return create_result(self.Xi, self.yi, self.space, self.rng,
                              models=self.models)
 
+    def _check_y_is_valid(self, x, y):
+        """Checks if y is valid (with respect to x in some cases)."""
+
+        if "ps" in self.acq_func:
+            if is_2Dlistlike(x):
+                if not (np.ndim(y) == 2 and np.shape(y)[1] == 2):
+                    raise TypeError("expcted y to be a list of (func_val, t)")
+            elif is_listlike(x):
+                if not (np.ndim(y) == 1 and len(y) == 2):
+                    raise TypeError("expected y to be (func_val, t)")
+
+        # if y isn't a scalar it means we have been handed a batch of points
+        elif is_listlike(y) and is_2Dlistlike(x):
+            for y_value in y:
+                if not isinstance(y_value, Number):
+                    raise ValueError("expected y to be a list a scalars")
```

Comment:
-> "expected y to be a list _of_ scalars not %s." % y

Not sure if we should display `y`, my reaction to exceptions like this is usually: "but y is a damn list of scalars!!!11!" so having it shown to me helps. But it could be a huge list ??

Reply:
How about "expected y to be a list of scalars but encountered 'z' at index 10"?

I thought about displaying the list partly (e.g. "[..., 1, 'z', 1,...]") but then the index of the invalid y_value is still unknown.

This is the check i added, there is a version for a single y value as well. If we change this, we should adapt the other error message as well.